js/inspector.ts

```diff
@@ -0,0 +1,153 @@
+                height: 32px;
+            }
+
+            .spacer {
+                width: 8px;
+            }
+
+            .object-browser {
+                max-height: 300px;
+                min-width: 290px;
+                overflow: scroll;
+            }
+
+            input[type='checkbox'] {
+                vertical-align: middle;
+            }
+        `,
+    ];
+
+    @property() hideCloseButton: boolean = false;
+    @property() expandPoints: boolean = false;
+    @property() expandPixels: boolean = true;
+    @property() expandObjects: boolean = false;
+    @property() pointInfo: Node = {};
+    @property() pixelInfo: Node = {};
+    @property() objectInfo: Node = {};
+
+    modelNameToViewName(): Map<keyof InspectorModel, keyof Inspector | null> {
+        return new Map([
+            ['hide_close_button', 'hideCloseButton'],
+            ['expand_points', 'expandPoints'],
+            ['expand_pixels', 'expandPixels'],
+            ['expand_objects', 'expandObjects'],
+            ['point_info', 'pointInfo'],
+            ['pixel_info', 'pixelInfo'],
+            ['object_info', 'objectInfo'],
+        ]);
+    }
+
+    render() {
+        return html`
+            <widget-container
+                .hideCloseButton="${this.hideCloseButton}"
+                @close-clicked="${this.onCloseButtonClicked}"
+            >
+                <div class="checkbox-container">
+                    <span class="legacy-text">Expand</span>
+                    <div>
+                        <input
+                            type="checkbox"
+                            .checked="${this.expandPoints}"
+                            @change="${this.onPointCheckboxEvent}"
+                        />
+                        <span class="legacy-text">Point</span>
+                    </div>
+                    <div>
+                        <input
+                            type="checkbox"
+                            .checked="${this.expandPixels}"
+                            @change="${this.onPixelCheckboxEvent}"
+                        />
+                        <span class="legacy-text">Pixels</span>
+                    </div>
+                    <div>
+                        <input
+                            type="checkbox"
+                            .checked="${this.expandObjects}"
+                            @change="${this.onFeatureCheckboxEvent}"
+                        />
+                        <span class="legacy-text">Objects</span>
+                    </div>
+                </div>
+                <div class="object-browser">
+                    ${this.renderNode(this.pointInfo)}
+                    ${this.renderNode(this.pixelInfo)}
+                    ${this.renderNode(this.objectInfo)}
+                </div>
+            </widget-container>
+        `;
+    }
+
+    private renderNode(node: Node): TemplateResult | typeof nothing {
+        if ((node.children?.length ?? 0) > 0) {
```

Comment:
nit: Simplify: `if (node.children?.length)`. Also can you document why this check is necessary?

Reply:
I didn't end up documenting it. Any reason in particular it should be documented?